parquet-hadoop/src/main/java/parquet/hadoop/InternalParquetRecordReader.java

```diff
@@ -102,13 +102,14 @@ public InternalParquetRecordReader(ReadSupport<T> readSupport, UnboundRecordFilt
   private void checkRead() throws IOException {
     if (current == totalCountLoadedSoFar) {
       if (current != 0) {
-        long timeAssembling = System.currentTimeMillis() - startedAssemblingCurrentBlockAt;
-        totalTimeSpentProcessingRecords += timeAssembling;
-        LOG.info("Assembled and processed " + totalCountLoadedSoFar + " records from " + columnCount + " columns in " + totalTimeSpentProcessingRecords + " ms: "+((float)totalCountLoadedSoFar / totalTimeSpentProcessingRecords) + " rec/ms, " + ((float)totalCountLoadedSoFar * columnCount / totalTimeSpentProcessingRecords) + " cell/ms");
-        long totalTime = totalTimeSpentProcessingRecords + totalTimeSpentReadingBytes;
-        long percentReading = 100 * totalTimeSpentReadingBytes / totalTime;
-        long percentProcessing = 100 * totalTimeSpentProcessingRecords / totalTime;
-        LOG.info("time spent so far " + percentReading + "% reading ("+totalTimeSpentReadingBytes+" ms) and " + percentProcessing + "% processing ("+totalTimeSpentProcessingRecords+" ms)");
+        totalTimeSpentProcessingRecords += (System.currentTimeMillis() - startedAssemblingCurrentBlockAt);
+        if (Log.INFO) {
+            LOG.info("Assembled and processed " + totalCountLoadedSoFar + " records from " + columnCount + " columns in " + totalTimeSpentProcessingRecords + " ms: "+((float)totalCountLoadedSoFar / totalTimeSpentProcessingRecords) + " rec/ms, " + ((float)totalCountLoadedSoFar * columnCount / totalTimeSpentProcessingRecords) + " cell/ms");
+            long totalTime = totalTimeSpentProcessingRecords + totalTimeSpentReadingBytes;
+            long percentReading = totalTime == 0 ? 0 : (100 * totalTimeSpentReadingBytes / totalTime);
+            long percentProcessing = totalTime == 0 ? 0 : (100 * totalTimeSpentProcessingRecords / totalTime);
+            LOG.info("time spent so far " + percentReading + "% reading ("+totalTimeSpentReadingBytes+" ms) and " + percentProcessing + "% processing ("+totalTimeSpentProcessingRecords+" ms)");
```

Comment:
Does this need to be logged if the total time processing and reading is 0ms? I think it would be better to do this:

```java
if (totalTime > 0) {
  long percentReading = ...;
  long percentProcessing = ...;
  LOG.info("time spent so far"...);
}
```